src/main.rs

```diff
@@ -154,6 +159,14 @@ fn main() {
             _ = remove_file(&args.downstream);
         });
 
+        // If the system booted with systemd,
+        // inform systemd that filterway is ready using notify.
+        // Other services that depend on filterway can start now.
+        if let Ok(sd_booted) = sd_notify::booted() {
+            println!("init detected as being systemd, notifying of readiness.");
```

Comment:
Would you mind making this debug-level only and use `eprintln`, also capitalize first letter (sorry, nit-picking here). I follow the convention of things you might want to use in a script or pipe (like json output) goes to stdout, all logging and other human-oriented stuff goes to stderr.